lib/request-state.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+var Map = require('immutable').Map;
+var History = require('./history');
+
+/**
+ * A Request State is an immutable state corresponding to a request processing or processed by the system.
+ *
+ * @param {string} status - The current status of the request
+ * @param {Map.<string, Job>} jobs - A map of all the jobs related to this request
+ * @param {History} history - A human-readable history for this request
+ * @param {Map.<string, *>} metadata - The metadata extracted by Specberus for this request
+ */
+function RequestState(status, jobs, history, metadata) {
+  if (typeof this !== 'object') {
+    throw new TypeError('RequestState must be constructed via new');
+  }
+
+  this.status = typeof status !== 'undefined' ? status : '';
```

Comment:
The `typeof` thing is only required for global vars (to avoid reference errors), so it's not needed here indeed. That said, it's not uncommon to see it done that way because double-safe habits die hard (I know it happens to me a fair bit).